dev/com.ibm.ws.jsonb_fat/fat/src/com/ibm/ws/jsonb/fat/FATSuite.java

```diff
@@ -113,8 +159,125 @@ public static void beforeSuite() throws Exception {
+            server.addEnvVar("JSONP_JAR", "fake-json-p.jar");
+
+            //TODO replace with a third party implementation when one is developed
+            server.addEnvVar("JSONB_PATH", "yasson/3.0.0-RC2/");
+            server.addEnvVar("JSONB_JAR", "yasson.jar");
+            server.addEnvVar("JSONB_ALT_JAR", "yasson.jar");
+
+            server.addEnvVar("YASSON_PATH", "yasson/3.0.0-RC2/");
+            server.addEnvVar("YASSON_JAR", "yasson.jar");
+        } else if (JakartaEE9Action.isActive()) {
+            server.addEnvVar("JSONP_PATH", "johnzon/1.2.18/jakarta/");
+            server.addEnvVar("JSONP_JAR", "johnzon-core.jar");
+
+            server.addEnvVar("JSONB_PATH", "johnzon/1.2.18/jakarta/");
+            server.addEnvVar("JSONB_JAR", "johnzon-jsonb.jar, johnzon-mapper.jar");
+            server.addEnvVar("JSONB_ALT_JAR", "johnzon-*.jar");
+
+            server.addEnvVar("YASSON_PATH", "yasson/2.0.4/");
+            server.addEnvVar("YASSON_JAR", "yasson.jar");
+        } else {
+            server.addEnvVar("JSONP_PATH", "johnzon/1.2.18/javax/");
+            server.addEnvVar("JSONP_JAR", "johnzon-core.jar");
+
+            server.addEnvVar("JSONB_PATH", "johnzon/1.2.18/javax/");
+            server.addEnvVar("JSONB_JAR", "johnzon-jsonb.jar, johnzon-mapper.jar");
+            server.addEnvVar("JSONB_ALT_JAR", "johnzon-*.jar");
+
+            server.addEnvVar("YASSON_PATH", "yasson/1.0.4/");
+            server.addEnvVar("YASSON_JAR", "yasson.jar");
+        }
+    }
+
+    /**
+     * Returns the jsonb provider class name that is available to the applications running
+     * based on the jakarta level being used.
+     *
+     * @param isAvailable - provide a false value if you want a provider class name that ISN'T available for negative testing.
+     * @return String - classname
+     */
+    public static String getJsonbProviderClassName(boolean... isAvailable) {
+        if (isAvailable.length > 1) {
+            throw new RuntimeException("Provide only one value for isAvailable (true/false)");
+        }
+
+        //If the length is 1 and set to false, provide the native case
+        if (isAvailable.length == 1 && !isAvailable[0]) {
+            return JakartaEE10Action.isActive() ? PROVIDER_JOHNZON_JONSB : PROVIDER_YASSON;
+        }
+
+        //In all other situations we should return the provider that is available
+        return JakartaEE10Action.isActive() ? PROVIDER_YASSON : PROVIDER_JOHNZON_JONSB;
+    }
+
+    /**
+     * Returns the jsonp provider class name that is available to the applications running
+     * based on the jakarta level being used.
+     *
+     * @param isAvailable - provide a false value if you want a provider class name that ISN'T available for negative testing.
+     * @return String - classname
+     */
+    public static String getJsonpProviderClassName(boolean... isAvailable) {
+        if (isAvailable.length > 1) {
```

Comment:
Is there a reason we want to allow someone to call this with multiple input parms and get an exception back instead of just having 2 methods with one of them taking a single boolean and the other like this:
public static String getJsonpProviderClassName(){
return Fatsuite.getJsonpProviderClassName(true);
}

Same comment apples to getJsonbProviderClassName

Reply:
So realistically I could have just made this a single boolean method where a "user" would provide true if they wanted the provider class name that should be loaded, and false if they wanted a provider class name that should NOT be loaded. 

But really the true case should be the default, and they should optionally be able to provide a false value if they want to perform negative tests. However, java doesn't support optional method parameters except for variable-length parameters (using the ... syntax).  So this is my shining example of how support for optional parameters could be useful in java.